cosmoz-grouped-list.js

```diff
@@ -215,11 +214,12 @@
 			if (!data[0].items) {
 				// no grouping, so render items as a standard list
 				this._groupsMap = null;
+				this.selectedItems = this.selectedItems.filter(i => data.indexOf(i) >= 0);
 				return data.slice();
 			}
 			this._groupsMap = new WeakMap();
 
-			return data.reduce((flatData, group) => {
+			const flatData = data.reduce((flatData, group) => {
```

Comment:
A risk for confusion that the aggregate and the outer const are named the same?
```suggestion
			const flattenedData = data.reduce((flatData, group) => {
```